OpenRA.Mods.Common/CampaignProgress.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using OpenRA.FileSystem;
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common
+{
+	public class CampaignProgress
+	{
+		private static string progressFile = Platform.ResolvePath("^", "cnc-progress.yaml");
```

Comment:
Mod specific static state causes problems, and this hardcodes cnc-specific logic in the common mod dll.